File: 4챕터/mostNumber/mostNumber.cpp
// N**2 알고리즘
int majority1(const vector<int>& A){
    int N = A.size();
    int majority = -1, majorityCount = 0;
    for(int i = 0; i < N; ++i){
        // A에 등장하는 A[i]의 수를 센다
        int V = A[i], count = 0;
        for(int j = 0; j < N; ++j){
            if(A[j] == V){
                ++count;
            }
        }

        // 현재 최대 빈도보다 많다면 갱신
        if(count > majorityCount){
            majorityCount = count;
            majority = V;
        }
    }
    return majority;
}

// N+range 알고리즘
int majority2(const vector<int>& A){
    int N = A.size();
    vector<int> count(101, 0);
    for(int i=0; i<N; ++i){
        count[A[i]]++;
    }
    int majority = 0;
    for(int i=1; i<=100; ++i){
        if(count[i] > count[majority]){
            majority = i;
        }
    }
    return majority
}

File: 4챕터/mostNumber/mostNumber.js
var mostNumber1 = function (numberArray) {
    var majority = -1, majorityCount = 0;
    numberArray.forEach(function (currentValue) {
        var count = 0;
        for (var i = 0; i < numberArray.length; i++) {
            if (numberArray[i] === currentValue) {
                count++;
            }
        }
        if (count > majorityCount) {
            majorityCount = count;
            majority = currentValue;
        }
    });
    return majority;
};
var test = [
    1, 3, 24, 13, 54, 65, 23, 45, 34, 77, 65, 2, 11, 2, 2, 2, 11, 3, 44, 5, 6, 6,
    5, 24,
];
console.log(mostNumber1(test));
